presto-mysql/src/main/java/com/facebook/presto/plugin/mysql/MySqlClient.java

```diff
@@ -115,32 +120,35 @@ protected SchemaTableName getSchemaTableName(ResultSet resultSet)
     @Override
     protected String toSqlType(Type type)
     {
-        if (Varchars.isVarcharType(type)) {
+        if (REAL.equals(type)) {
+            return "float";
+        }
+        if (TIME_WITH_TIME_ZONE.equals(type)) {
+            return "time";
+        }
+        if (TIMESTAMP.equals(type) || TIMESTAMP_WITH_TIME_ZONE.equals(type)) {
+            return "datetime";
```

Comment:
I think this conversion is wrong. Mysql `datetime` behaves like TIMESTAMP WITHOUT TIME ZONE should (according to standard), so it's suitable counterpart of our TIMESTAMP (when it's fixed).
TIMESTAMP WITH TIME ZONE represents a pair `(point in time, time zone)` (like `ZonedDateTime`) and I see no equivalent in MySQL. MySQL's `TIMESTAMP` seems to represent a point in time (like `Instant`), but no time zone.

Reply:
Agreed. This was originally just refactoring, but since we're touching this code, and we know it's broken, I'll add another commit to remove it.

Reply:
I understand. Either remove or add a comment warning it's broken (with preference on the first of course :) )